dev/sbt-plugin/src/main/scala/com/lightbend/lagom/sbt/LagomPlugin.scala

```diff
@@ -486,17 +487,24 @@ private[sbt] object ConsoleHelper {
           override def close(): Unit = services.foreach(_.close())
         })
       case _ =>
+        import scala.concurrent.ExecutionContext.Implicits.global
+        import scala.concurrent.duration._
         // blocks until user press CTRL+D
         interaction.waitForCancel()
         // then shut down all running services
         log.info("Stopping services")
-        services.foreach { service =>
-          print(".")
-          service.close()
-        }
+
+        //Stop services in asynchrone manner
+        val closing = Future.traverse(services)(serv => Future {
+          serv.close()
+        })
+        closing.onComplete(_ => println("All services are stopped"))
+        Await.result(closing, 60 seconds)
+
```

Comment:
This may be prone to deadlocks.  You're using the Scala global execution context.  Let's say there are 100 services.  Let's say the global execution context has 8 threads in it, you're going to shutdown 8 services at a time.  If those 8 services, during their shutdown, attempt to use the global execution context too, they'll be blocked, because it's exhausted.  And since their shutdown is now blocked, the threads being here will be permanently blocked.

You probably need to use a dedicated thread pool here to avoid that - and make sure you've shut it down after use.

The best solution here would be to modify Play's ServerWithStop so that it's stop() method returns a CompletionStage (note, it has to be a CompletionStage, not a Scala future, because this class is shared between classloaders that contain different versions/instances of Scala, which limits you to using classes built into the JDK).  That of course will take a while because it will require a new Play major release, but if you'd be interested in getting the ball rolling on that with a PR on Play, that would be great.